current.go

```diff
@@ -61,32 +60,71 @@ func (r *VaultRole) Search(config SearchConfig) (Role, error) {
 		return Role{}, microerror.Maskf(notFoundError, "no vault secret at path '%s'", key.RoleName(config.ID, config.Organizations))
 	}
 
-	b, err := json.Marshal(secret.Data)
+	role, err := vaultSecretToRole(secret)
 	if err != nil {
 		return Role{}, microerror.Mask(err)
 	}
 
-	var internalRole role
-	err = json.Unmarshal(b, &internalRole)
-	if err != nil {
-		return Role{}, microerror.Mask(err)
+	role.ID = config.ID
+	return role, nil
+}
+
+// vaultSecretToRole makes required type casts / type checks and parsing to
+// extract role information from Vault api.Secret.
+func vaultSecretToRole(secret *api.Secret) (Role, error) {
```

Comment:
I know this function looks horrible compared to old JSON trick, but I cannot figure any other way to make this robust and adaptable for both old and new Vault version.

Reply:
I think this is a perfect function for unit tests.

Reply:
Will check this. Input `api.Secret` must come from Vault in order for it to make sense. I'm not familiar with Vault API, but I'll investigate here.